### ISSUE TITLE
Fitting ontogeny including variability using PK-Sim Ontogeny Database approach

### ISSUE THREAD

**Issue by choitingy** [Open-Systems-Pharmacology/Forum#579]
Hi all,

I am trying to recreate the approach of fitting ontogeny including variability used in the PK-Sim Ontogeny Database.
 
I found the documentation for v7.3 here: https://github.com/Open-Systems-Pharmacology/OSPSuite.Documentation/blob/master/PK-Sim%20Ontogeny%20Database%20Version%207.3.pdf
 
My proposal to recreate these methods:
 
1. Fit the Hill equation (bottom of page 4) to the literature data. This line doubles as the median of pop fit and the geoMean A across PMA. Some adjustment to the starting point on x-axis may need to be made to allow better fitting (i.e., a higher n). 
2. Create upper and lower bounds defined by Hill equations with coefficient "k" to allow A>1 an A<1 to capture the observed data. Simulate 10,000 individuals with randomly selected k, A0.5, and n within the defined upper and lower bounds. 
3. Check whether 1.645x geoSD of the fitted line of geoMean A indeed capture 90% of the simulated individuals (50th to 95th percentile above the median pop fit, and 5th to 50th percentile below the median pop fit). If it does not, then try different upper and lower bounds (repeat step 2).
4. Once 90% of the simulated individuals are captured within the bounds, record the geoSD n, geoSD A0.5, and geoSD adult obtained from this population. 

However, I received a second opinion on the above approach and we came to conclusion that it was more likely that the bounds were not defined in such a prescribed way; and k, A0.5, and n were likely not chosen at random within those bounds. Based on the following statement in the documentation:
 
"For fitting the ontogeny including variability, constituting variability in ontogeny as well as activity, a virtual population with 10000 individuals was created". 
 
It sounds like the virtual population was made based on variability in ontogeny and variability in activity. These variabilities could have been drawn from normal distributions with some SD, but I am unsure as to whether it would be literature or another criteria informing these.
 
I would greatly appreciate if anyone has more insight on recapitulating the documentation methods.
 
One further question: I want to confirm that "SD" in the PK-Sim Table is the "geoSD around the fitted line" (it appears to be so judging by CYP3A5 and UGT1A4 in liver). I also ask this because most of the SDs in the tablet for younger ages are higher than later ages and I have difficulty corresponding that to the plots visually - see CYP2C8 in liver as an example.
 
Thanks!
Cindy

**Comment by prvmalik** [comment]
> One further question: I want to confirm that "SD" in the PK-Sim Table is the "geoSD around the fitted line" (it appears to be so judging by CYP3A5 and UGT1A4 in liver). I also ask this because most of the SDs in the tablet for younger ages are higher than later ages and I have difficulty corresponding that to the plots visually - see CYP2C8 in liver as an example.

Yes SD in the table represents geoSD in total activity/expression. GeoSD has a proportional relationship to the geomean value, rather than additive (in the case of arithmetic SD). Even with higher geoSDs at young ages, the final upper and lower bounds are smaller than at older ages because the absolute age value (e.g. 40 weeks PMA) is quite low.

**Comment by choitingy** [comment]
@prvmalik Thanks for the explanation on that Paul, it now makes sense to me.

Interested in hearing if anyone has feedback on the fitting approach I mentioned or any further insight.

**Comment by Yuri05** [comment]
>Interested in hearing if anyone has feedback on the fitting approach I mentioned or any further insight.

Maybe @Incei ?

**Comment by Incei** [comment]
@choitingy may I ask why you want to change the approach to fit the variability of the ontogeny profiles? 
Indeed the Geomean and GeoSD of the ontogenies were estimated based on reported variability in ontogeny and variability in activity in the literature that are referenced for each enzyme in the documentation.

**Comment by choitingy** [comment]
@Incei Not so much to change the approach, but rather to employ similar methods used in the PK-Sim documentation where possible. Essentially, I am building a model for a drug that is metabolized by UGT1A4 and would like to update the existing ontogeny profile by incorporating data from a recently published study.

Thanks for your clarification. Just as an example, for UGT1A6 in the liver it appears that the serotonin glucuronidation data from Figure 2A from Miyagi & Collier, 2011 (doi: 10.1124/dmd.110.037192) were digitized and used for the fitting. Would the used reported variability in ontogeny and variability in activity be found from Figure 2A or were the estimates also informed by elsewhere in this paper?

**Comment by Incei** [comment]
@choitingy, thanks, that sounds great. It would be good to always update/evaluate with new available data.

Regarding your question on>  Would the used reported variability in ontogeny and variability in activity be found from Figure 2A or were the estimates also informed by elsewhere in this paper?

Im not sure if I understood this correctly, but indeed only the extracted data from figure 2A have been used for fitting, In case you want to re-extract for your purpose, there should be free software available to do this. In the results section, there is no table as such available, other than birth and adults activity levels.

**Comment by moPellowe** [comment]
@Incei, I also want to understand the Methods in the ontogeny database documentation (as in the original post), but I am not clear on all of the details. The methods in the documentation says the following:

"For fitting the ontogeny including variability, constituting variability in ontogeny as well as activity, a virtual population with 10000 individuals was created. Fitted were then the geometric mean and geometric standard deviation of the Hill coefficient (GeoSD n), the geometric mean and geometric standard deviation of PMA at 50% activity compared to adult (GeoSD A0.5), and a geometric standard deviation as an activity variability factor (GeoSD adult)."

1) So what variability is used to build the virtual population? Is there any variation in enzyme expression or concentration in these individuals? Does the virtual population use the Hill equation that is fit to the ontogeny data?
2) I don't understand how the geometric mean and geometric standard deviation of the parameters is fit. Was a Hill function first fit to each individual, then the geometric mean and geometric standard deviations were calculated from that distribution of parameters?

**Comment by prvmalik** [comment]
> So what variability is used to build the virtual population? Is there any variation in enzyme expression or concentration in these individuals? Does the virtual population use the Hill equation that is fit to the ontogeny data

The Tables in the documentation are uploaded into PKSim. I believe that PK-Sim linearly interpolates between the points in the table. This interpolated function outputs the "Ontogeny factor" as a function of age. The geometric standard deviation is also inputted into the table and describes the variability for that age bin around the mean ontogeny factor. When an individual is generated, the individual receives a lognormally-distributed ontogeny factor using the mean and geoSD information from the interpolated function and table. The ontogeny factor is used to get enzyme concentration in the following equation:

Final Enzyme Concentration = Reference Enzyme Concentration * Organ Relative Expression * Ontogeny Factor

I'd also like to know the answer to point 2.

**Comment by moPellowe** [comment]
So @prvmalik, the tables in the documentation were used as input for the creation of the virtual population? Then the tables come from taking the geometric mean and geometric standard deviations of experimental enzyme activity results?

**Comment by prvmalik** [comment]
Yes. But as I'm learning it's rather difficult to get a geometric mean line from an optimization. It's either the mean or the median line, and the median is a decent approximation of the geometric mean.

**Comment by Incei** [comment]
> I don't understand how the geometric mean and geometric standard deviation of the parameters is fit. Was a Hill function first fit to each individual, then the geometric mean and geometric standard deviations were calculated from that distribution of parameters?

The created population of 10000 individuals had the following parameters: 
•	Age (range depending on available data)
•	Random number (between 0-1)  for calculation of the Hill coefficient
•	Random number  (between 0-1) for calculation of PMA
•	Random number (between 0-1) for calculation of  activity variability factor

In each step, new distributions were created using the random numbers of the population and the geo-mean and geo-sd of the fitparameters (hill function). These ontogeny distributions over age were than compared to the literatur data to evaluate the loss function [(wiki).](https://en.wikipedia.org/wiki/Loss_function#:~:text=In%20mathematical%20optimization%20and%20decision,cost%22%20associated%20with%20the%20event.) The best parameter fit was then used to create the ontogeny table.

Note: In case of an update, or new ontogeny function, we are not applying this method anymore. E.g. for AGP we applied MCMC, which can also be found in this documentation.

**Comment by moPellowe** [comment]
Thank you, @Incei, for that explanation. I have a few follow-up questions:

1. - So a new virtual population was created at each step of the fitting process? 
2. - And each individual was given a random value for the Hill coefficient [n], the PMA at 50% activity [A_0.5], and the activity variability factor on the entire expression [adult]? 
3. - Why would the random numbers generated for each of these three parameters be limited to the range [0,1]?
